src/joystick/joystick_manager.h

```diff
@@ -0,0 +1,96 @@
+        { Xbox360Controller::A, 0 },
+        { Xbox360Controller::B, 1 },                // Turn counterclockwise
+        { Xbox360Controller::X, 2 },                // Turn clockwise
+        { Xbox360Controller::Y, 3 },                // Enables ID switching and profile switching as long as it is pressed
+        { Xbox360Controller::LeftBumper   , 4  },   // Dribbler
+        { Xbox360Controller::RightBumper  , 5  },   // Kicker
+        { Xbox360Controller::Back         , 6  },   // +Y : Switch control mode (Fifa/CoD)
+        { Xbox360Controller::Start        , 7  },   // +Y : Switch profile
+        { Xbox360Controller::Guide        , 8  },
+        { Xbox360Controller::LeftStick    , 9  },
+        { Xbox360Controller::RightStick   , 10 }
+};
+
+// ======================================================================== //
+
+
+struct JoyEntry {
+    JoyEntry() : robotID{intSupplier}, MY_ID{intSupplier++} {
+        this->profile = profile_children;
+    }
+    void init();
+    void setToInput(std::string newInput);
+    int getTimer();
+    void resetTimer();
+    void receiveJoyMsg(const sensor_msgs::JoyConstPtr &msg);
+    void setRobotID(int id);
+    void nextJoystickProfile();
+    void switchControlMode();
+    void setControllerConnected(bool isConnected);
+
+    ::boost::optional<::boost::process::child> process;     // Holds the joy_node process
+    ::boost::optional<ros::Subscriber> subscriber;          // Subscribes to the joy_node topic (/js0, /js1 etc etc)
+    ::boost::optional<sensor_msgs::Joy> msg;                // Holds the latest message from the joy_node topic
+    ::boost::optional<sensor_msgs::Joy> msg_prev;           // Holds the latest message from the joy_node topic
+
+    std::string input;                          // js0
+    int robotID;                                // 1 - 16
+    const int MY_ID;                            // Holds the unique id
+
+    joystick_profile profile;
+    int profileCounter;
+
+    // ==== Variables related to automatically running a RoleNode
+    ::boost::optional<::boost::process::child> processAuto;         // Holds the joy_node auto process
+    bool dribblerOn = false;
+
+    // ==== Variables related to driving ==== //
+    rtt::Vector2 speedState;                    // Holds the x-speed and y-speed of the robot.
+    int genevaState;                            // Holds the state of the Geneva Drive. Range is [1,5]
+    bool controllerConnected = true;            // Holds if the corresponding controller is connected
+    float orientation = 0.0;                    // Holds the last orientation of the robot
+    float orientationOffset = 0.0;              // Holds the orientation offset
+    bool useRelativeControl = true;             // Holds the control mode (relative or absolute)
+    static int intSupplier;                     // Supplies ids to new instances of JoyEntry
+};
+
+template<typename T> T getVal(const std::vector<T> &values, int index);
```

Comment:
I don't like this function... this is called Templating and it is extremely overkill

Reply:
Yeah I had a discussion about this with Baris but we didn't really come to a solution? What do you think would be a better way to do it?

Reply:
So... the button returns a value, which is always an integer, right? So the function could at least be rewritten to fit only integers. Furthermore, something like isPressed would be even more awesome

Reply:
The value can also get floats as input, and hence return a float, so that won't work